pkg/ssh/manager.go

```diff
@@ -15,6 +19,47 @@ type StreamManager struct {
 	mu            sync.Mutex
 	cfg           *config.Config
 	activeStreams map[uint64]*StreamHandler
+
+	listenersMu sync.Mutex
+	listeners   map[string][]func(*session.Session)
+}
+
+// ClearRecords implements databroker.SyncerHandler.
+func (sm *StreamManager) ClearRecords(ctx context.Context) {
+}
+
+// GetDataBrokerServiceClient implements databroker.SyncerHandler.
+func (sm *StreamManager) GetDataBrokerServiceClient() databroker.DataBrokerServiceClient {
+	return sm.auth.GetDataBrokerServiceClient()
+}
+
+// UpdateRecords implements databroker.SyncerHandler.
+func (sm *StreamManager) UpdateRecords(ctx context.Context, serverVersion uint64, records []*databroker.Record) {
+	sm.listenersMu.Lock()
+	defer sm.listenersMu.Unlock()
+	for _, record := range records {
+		if record.DeletedAt == nil {
+			continue
+		}
+		var s session.Session
+		if err := record.GetData().UnmarshalTo(&s); err != nil {
+			panic(err)
+		}
+		for _, fn := range sm.listeners[record.Id] {
+			fn(&s)
+		}
+		sm.listeners[record.Id] = []func(*session.Session){}
```

Comment:
This should probably be `delete(sm.listeners, record.Id)`? Otherwise its a memory leak?